lib/src/gitignore.rs

```diff
@@ -60,22 +64,31 @@ impl GitIgnoreFile {
     pub fn chain(
         self: &Arc<GitIgnoreFile>,
         prefix: &str,
+        ignore_path: &Path,
         input: &[u8],
     ) -> Result<Arc<GitIgnoreFile>, GitIgnoreError> {
         let mut builder = gitignore::GitignoreBuilder::new(prefix);
         for (i, input_line) in input.split(|b| *b == b'\n').enumerate() {
             let line =
                 std::str::from_utf8(input_line).map_err(|err| GitIgnoreError::InvalidUtf8 {
-                    path: PathBuf::from(prefix),
+                    path: ignore_path.to_path_buf(),
                     line_num_for_display: i + 1,
                     line: String::from_utf8_lossy(input_line).to_string(),
                     source: err,
                 })?;
             // FIXME: do we need to provide the `from` argument? Is it for providing
             // diagnostics or correctness?
```

Comment:
Maybe this comment can be removed? (or maybe we can pass it to `.add_line(Some(ignore_path), ..)`?

Reply:
I looked through the `ignore` crate, and it looks like it's neither for diagnostics nor correctness. It just stores it in the `Glob` so that when you call `GitIgnore::matched` or `GitIgnore::matched_path_or_any_parents`, it can tell you which file the matching glob came from. We ignore the `Glob` returned from these methods entirely, so I think it should be safe to remove the comment. I don't think we should pass it the path, since it causes an unnecessary clone of the string.